File: 48-rotate-image/48-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        
        int m = matrix.size();
        int n = matrix[0].size();
        
        vector<vector<int>> temp(m, vector<int>(n));
        
        for(int i = 0; i < m; i++) 
        {
            for(int j = 0; j < n; j++)
            {
                temp[j][n - i - 1] = matrix[i][j];
            }
        }
        
        matrix = temp;
        
    }
};

File: 48-rotate-image/48-rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        
        int n = matrix.length;
        
        // transpose
        
        for(int i = 0; i < n; i++)
        {
            for(int j = 0; j < i; j++)
            {
                int temp = 0;
                temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
            }
        }
        
        // reverse
        
        for(int i = 0; i < n; i++)
        {
            for(int j = 0; j < n/2; j++)
            {
                int temp = 0;
                temp = matrix[i][j];
                matrix[i][j] = matrix[i][matrix.length - j - 1];
                matrix[i][matrix.length - j - 1] = temp;
            }
        }
    }
}